lib/src/Uncertainty/Algorithm/Simulation/SubsetSampling.cxx

```diff
@@ -0,0 +1,572 @@
+: Simulation(event)
+, proposalRange_(proposalRange)
+, targetProbability_(targetProbability)
+, iSubset_(false)
+, betaMin_(ResourceMap::GetAsNumericalScalar("SubsetSampling-DefaultBetaMin"))
+, keepEventSample_(false)
+, numberOfSteps_(0)
+{
+  setMaximumOuterSampling(ResourceMap::GetAsUnsignedInteger("SubsetSampling-DefaultMaximumOuterSampling"));// overide simulation default outersampling
+  UnsignedInteger outputDimension = event.getFunction().getOutputDimension();
+  if (outputDimension > 1)
+    throw InvalidArgumentException(HERE) << "Output dimension for SubsetSampling cannot be greater than 1, here output dimension=" << outputDimension;
+}
+
+
+/* Virtual constructor */
+SubsetSampling * SubsetSampling::clone() const
+{
+  return new SubsetSampling(*this);
+}
+
+
+/* Performs the actual computation. */
+void SubsetSampling::run()
+{ 
+  // First, initialize some parameters
+  convergenceStrategy_.clear();
+  numberOfSteps_ = 0;
+  thresholdPerStep_.clear();
+  gammaPerStep_.clear();
+  coefficientOfVariationPerStep_.clear();
+  probabilityEstimatePerStep_.clear();
+  eventInputSample_.clear();
+  eventOutputSample_.clear();
+
+  dimension_ = getEvent().getAntecedent()->getDimension();
+
+  if (getMaximumCoefficientOfVariation() != ResourceMap::GetAsNumericalScalar("Simulation-DefaultMaximumCoefficientOfVariation"))
+    Log::Warn(OSS() << "The maximum coefficient of variation was set. It won't be used as termination criteria.");
+
+  if (targetProbability_ * getMaximumOuterSampling() < 1)
+    throw InvalidArgumentException(HERE) << "maximumOuterSampling (" << getMaximumOuterSampling() << ") should be >= " << ceil(1.0 / targetProbability_);
+
+  if (getMaximumOuterSampling() * getBlockSize() <= 100)
+    Log::Warn(OSS() << "The number of samples per step is very low : " << getMaximumOuterSampling()*getBlockSize() << ".");
+
+  // perform isoprobabilistic transformation (the study is done in the standard space):
+  standardEvent_ = StandardEvent(getEvent());
+
+  NumericalScalar currentCoVsquare = 0.0;
+  NumericalScalar varianceEstimate = 0.0;
+  NumericalScalar coefficientOfVariationSquare = 0.0;
+
+  // allocate input/output samples
+  const UnsignedInteger maximumOuterSampling = getMaximumOuterSampling();
+  const UnsignedInteger blockSize = getBlockSize();
+  const UnsignedInteger N = maximumOuterSampling * blockSize;
+  currentPointSample_ = NumericalSample(N, dimension_);
+  currentLevelSample_ = NumericalSample(N, getEvent().getFunction().getOutputDimension());
+  
+  // Step 1: sampling
+  for (UnsignedInteger i = 0; i < maximumOuterSampling; ++ i)
+  {
+    NumericalSample inputSample;
+    if (!iSubset_) {
+      // crude MC
+      inputSample = standardEvent_.getAntecedent()->getDistribution().getSample(blockSize);
+    }
+    else {
+      // conditional sampling
+      TruncatedDistribution truncatedChiSquare(ChiSquare(dimension_), betaMin_ * betaMin_, TruncatedDistribution::LOWER);
```

Comment:
Here you need a truncated Chi distribution, not a truncated ChiSquare distribution. The lower bound is betaMin_.